test/utilities.js

```diff
@@ -14,7 +17,56 @@ let toHex = function(buffer) {
     return '0x' + buffer;
 };
 
+let generateMerkleRootAndProof = function(leaves, txIndex) {
+    return generateMerkleRootAndProofHelper(leaves, 16, txIndex, 0);
+};
+
+let generateMerkleRootAndProofHelper = function(leaves, depth, txIndex, zeroHashesIndex) {
```

Comment:
could you add some documentation on what exactly this helper method is doing?

Reply:
Added comments